site/src/Controller/UserController.php

```diff
@@ -223,4 +228,143 @@ public function resendConfirmationEmailAction(
+        );
+        $count = $notificationManager->getUnseenNotificationCount($notifiable);
+
+        return new JsonResponse($count);
+    }
+
+    /**
+     * View all notifications in the user's inbox
+     *
+     * @Route("/user/notification/inbox", name="notification_inbox")
+     *
+     * @param NotificationManager $notificationManager
+     *
+     * @return string;
+     */
+    public function inboxAction(NotificationManager $notificationManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+        $user = $this->getUser();
+        $notifications = $notificationManager->getNotifications($user);
+
+        return $this->render(
+            'user/notification_inbox.html.twig',
+            [
+                'notificationList' => $notifications,
+            ]
+        );
+    }
+
+    /**
+     * Mark a notification as seen in the notification inbox
+     *
+     * @Route("/user/notification/seen/{notification}", name="inbox_mark_seen")
+     *
+     * @param Notification        $notification
+     * @param NotificationManager $notificationManager
+     *
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws \Doctrine\ORM\EntityNotFoundException
+     * @throws \Exception
+     *
+     * @return Response
+     */
+    public function inboxMarkSeenAction($notification, NotificationManager $notificationManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        $user = $this->getUser();
+
+        $notifiableEntity = $notificationManager->getNotifiableEntity($user);
+        $notifiable = $notificationManager->getNotifiableInterface($notificationManager->getNotifiableEntityById($notifiableEntity));
+
+        $notificationManager->markAsSeen(
+            $notifiable,
+            $notificationManager->getNotification($notification),
+            true
+        );
+        $notificationList = $notificationManager->getNotifications(
+            $notifiable
+        );
+
+        return $this->redirectToRoute(
+            'notification_inbox',
+            ['notificationList' => $notificationList]
+        );
+    }
+
+    /**
+     * Remove a Notification
+     *
+     * @Route("/user/notification/remove/{notification}", name="remove_notification")
+     *
+     * @param Notification        $notification
+     * @param NotificationManager $notificationManager
+     *
+     * @throws \Doctrine\ORM\OptimisticLockException
+     *
+     * @return Response
+     */
+    public function removeAction($notification, NotificationManager $notificationManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        $user = $this->getUser();
+
+        $notifiableEntity = $notificationManager->getNotifiableEntity($user);
+        $notifiable = $notificationManager->getNotifiableInterface($notificationManager->getNotifiableEntityById($notifiableEntity));
+
+
+        // Deletes the link between notification and notifiable
```

Comment:
I think I asked this before: Why don't take foreign constraints care of deleting all parts of a notification? If they do (as they should), you should be OK with only one remove* function call. Having only one such call is really helpful: now your remove becomes a single transaction and the database takes care that no intermediate state becomes visible (e.g. a notification being half-deleted).

Reply:
@imphil We will resolve this discuss in the meeting today. It's a bit tricky one!

Reply:
Added a comment in the code as discussed